Docs: updated `README.rst` file (#563)
googleapis/python-spanner-django: README.rst

<https://docs.djangoproject.com/en/2.2/ref/databases/#using-a-3rd-party-database-backend>`__
for Django.

How it works
------------

Overall design
~~~~~~~~~~~~~~

.. figure:: ./assets/overview.png
   :alt:

Internals
~~~~~~~~~

.. figure:: ./assets/internals.png
   :alt:

Installation
------------

Using this library requires a Google Cloud Platform project with the Cloud

> Spanner API enabled. See the Spanner' Python Client `documentation

Reviewer: ```suggestion
Spanner API enabled. See the Spanner Python client `documentation
```

I'm no usage expert, but I think "Spanner" and "Python" are capitalized here as separate terms -- not as part of "Spanner Python Client" -- and "client" can be lowercase.

Author: 👍 Done.

    pip3 install django-google-spanner

> To install from the source:

Reviewer: ```suggestion
To install from source:
```

"Install from source" actually sounds more natural to me. I was curious, but don't know that this proves anything:

![image](https://user-images.githubusercontent.com/43799105/101409353-a1734980-3892-11eb-9ffa-91046588955a.png)

Author: 👍 Done!

           ...
       ]

-  Edit the ``DATABASES`` settings to point to an EXISTING database, as shown in the following example:

   .. code:: python

       DATABASES = {
           'default': {
               'ENGINE': 'django_spanner',
               'PROJECT': '<GCP_project_id>',
               'INSTANCE': '<instance_id>',
               'NAME': '<database_name>',
           }
       }

> -   In order to retrieve the Cloud Spanner credentials from a JSON file, ``credentials_uri`` parameter can also be supplied in the ``OPTIONS`` field:

Reviewer: ```suggestion
-   In order to retrieve the Cloud Spanner credentials from a JSON file, the ``credentials_uri`` parameter can also be supplied in the ``OPTIONS`` field:
```

Author: Done!

models by ID isn't guaranteed to return them in the order in which they
were created.

``ForeignKey`` constraints aren't created (`#313 <https://github.com/googleapis/python-spanner-django/issues/313>`__)
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Spanner does not support ``ON DELETE CASCADE`` when creating foreign-key

> constraints, so these are not supported in ``django-google-spanner``.

Reviewer: ```suggestion
constraints, so this is not supported in ``django-google-spanner``.
```

Author: 👍 Done!

Check constraints aren't supported
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> Spanner does not support ``CHECK`` constraints, so one isn't created for

Reviewer: "Doesn't" get expanded by "isn't" doesn't?

Author: 👍 Done!

Schema migrations
~~~~~~~~~~~~~~~~~

> There are some limitations on schema changes to consider:

Reviewer: 👍

Author: Looks like this one has already been fixed.
